sn_networking/src/event/kad.rs

```diff
@@ -457,6 +506,64 @@ impl SwarmDriver {
         Ok(())
     }
 
+    /// Checks passed peers from a request and checks they are sufficiently spaced to
+    /// ensure we have searched enough of the network range as determined by our `get_range`
+    fn have_we_have_searched_thoroughly_for_quorum(
+        expected_get_range: KBucketDistance,
+        searched_peers_list: &HashSet<PeerId>,
+        data_key_address: &NetworkAddress,
+        quorum: &Quorum,
+        // if it's the last step, we can be more lenient with the quorum as we've searched
+        // as far as we can
+        is_last_step: bool,
+    ) -> bool {
+        warn!("Assessing search: range: {:?}, address: {data_key_address:?}, quorum required: {quorum:?}, peers_returned_count: {:?}", expected_get_range.ilog2(), searched_peers_list.len());
+
+        let required_quorum = get_quorum_value(quorum);
+
+        // if we don't even meet quorum, that's it.
+        if !is_last_step && searched_peers_list.len() < required_quorum {
+            return false;
+        }
+
+        // get the farthest distance between peers in the response
+        let mut current_distance_searched = KBucketDistance::default();
```

Comment:
the following block of code seems just set `current_distance_searched` to be the `searched closet distance to the target` ?

Reply:
🤔  


```rust
   // get the farthest distance between peers in the response
        let mut current_distance_searched = KBucketDistance::default();

        // iterate over peers and see if the distance to the data is greater than the get_range
        for peer_id in searched_peers_list.iter() {
            let peer_address = NetworkAddress::from_peer(*peer_id);
            let distance_to_data = peer_address.distance(data_key_address);
            if current_distance_searched < distance_to_data {
                current_distance_searched = distance_to_data;
            }
        }
```

Should be:

- set `current_distance_searched` = 0
- if the peer's distance to data is further, use that distance.

Which I think isn't only going to be the closest?? (Or i'm missing something).


Thinking about that now though, the distasnce searched may yet be bigger... as the peers could be either side... 

ie, 
- Peer1: Location =1
- Data: Location = 2
- Peer2: Location = 3

So the distance of any peer is 1... But the total searched is 2...

🤔 I'm not sure how best to calculate _that_ in libp2p address terms 🤔

Reply:
I think we just need to sort to a fixed point. Ideally a distance of 0. But it's all relative, right. So long as we choose _somewhere consistent_. 

I've changed this to derive an address from the genesis spend just now. It would be consistent across the whole network then... 

(Although I'm not actually sure there's any benefit of that from our_peer_id eg... )

Reply:
Actually that's bad and leads to wild distances. I guess we just want the farthest distance to the key we searched for or something similar. Feels simpler